src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/processorders/ProcessOrdersAboutToSubmitHandlerTest.java

```diff
@@ -62,6 +76,112 @@ void testCanHandle() {
+                        .uploadDraftDocument(TARGET_DOCUMENT_1).build()).build(),
+                    DirectionOrderCollection.builder().value(DirectionOrder.builder().originalDocument(TARGET_DOCUMENT_2)
+                        .uploadDraftDocument(TARGET_DOCUMENT_2).build()).build(),
+                    expectedNewDirectionOrderCollection = DirectionOrderCollection.builder().value(DirectionOrder.builder()
+                        .uploadDraftDocument(TARGET_DOCUMENT_3).build()).build()
+                ))
+                .build())
+            .build());
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> res = underTest.handle(finremCallbackRequest, AUTH_TOKEN);
+
+        assertThat(res.getData().getUploadHearingOrder()).hasSize(2);
+        assertThat(res.getData().getUploadHearingOrder().get(1)).isEqualTo(expectedNewDirectionOrderCollection);
+    }
+
+    @Test
+    void shouldStampNewUploadedDocumentsFromUnprocessedApprovedDocuments() throws JsonProcessingException {
+        List<DirectionOrderCollection> uploadHearingOrder = new ArrayList<>(List.of(
+            DirectionOrderCollection.builder().value(DirectionOrder.builder().uploadDraftDocument(TARGET_DOCUMENT_4).build()).build()
+        ));
+        DirectionOrderCollection expectedNewDirectionOrderCollection = null;
+
+        FinremCallbackRequest finremCallbackRequest = FinremCallbackRequestFactory.from(FinremCaseData.builder()
+            .uploadHearingOrder(uploadHearingOrder)
+            .draftOrdersWrapper(DraftOrdersWrapper.builder()
+                .unprocessedApprovedDocuments(List.of(
+                    DirectionOrderCollection.builder().value(DirectionOrder.builder().originalDocument(TARGET_DOCUMENT_1)
+                        .uploadDraftDocument(TARGET_DOCUMENT_1).build()).build(),
+                    DirectionOrderCollection.builder().value(DirectionOrder.builder().originalDocument(TARGET_DOCUMENT_2)
+                        .uploadDraftDocument(TARGET_DOCUMENT_2).build()).build(),
+                    expectedNewDirectionOrderCollection = DirectionOrderCollection.builder().value(DirectionOrder.builder()
+                        .uploadDraftDocument(TARGET_DOCUMENT_3).build()).build()
+                ))
+                .build())
+            .build());
+
+        // Captor to capture the argument passed to the service
+        ArgumentCaptor<FinremCaseDetails> caseDetailsCaptor = ArgumentCaptor.forClass(FinremCaseDetails.class);
+
+        // Use InOrder to verify the order of method calls
+        InOrder inOrder = inOrder(underTest, additionalHearingDocumentService);
+
+        // verify handleNewDocument should be invoked before createAndStoreAdditionalHearingDocuments which stamps the PDF file uploaded.
+        underTest.handle(finremCallbackRequest, AUTH_TOKEN);
+
+        inOrder.verify(underTest).handleNewDocument(any(FinremCaseData.class));
```

Comment:
You shouldn't need to test the order in which internal methods are called. All that matters is that the correct data is passed to the additionalHearingDocumentService dependency